### ISSUE TITLE
cannot create a Calendar/Contacts

### ISSUE THREAD

**Issue by cremesk** [modoboa/modoboa#1656]
Hallo, i have a Fresh default Installed Modoboa Version. It works very nice but a have a liddle Problem to create a Calendar for my User.

# Impacted versions
OS: Debian 9.6 (stretch)

* Modoboa: 1.13.0
* installer used: Yes
* Webserver: Nginx

**Comment by tonioo** [comment]
@cremesk Are you using a valid TLS certificate? (letsencrypt)

**Comment by cremesk** [comment]
@tonioo yes i do. https://mail.secnux.net/
demo / Password123

**Comment by tonioo** [comment]
@cremesk Can you set DEBUG to True inside settings.py and reload uwsgi ?

**Comment by cremesk** [comment]
@tonioo okay Done. The Calendar Page has an IO Error. You can Login the Demo Account an check it.
```

Request Method: | GET
-- | --
https://mail.secnux.net/calendars/
1.11.18
IOError
Error reading /srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/../frontend/webpack-stats.json. Are you sure webpack has generated the file and the path is correct?
/srv/modoboa/env/local/lib/python2.7/site-packages/webpack_loader/loader.py in _load_assets, line 32
/usr/bin/uwsgi-core
2.7.13
['.',  '',  '/srv/modoboa/env/lib/python2.7',  '/srv/modoboa/env/lib/python2.7/plat-x86_64-linux-gnu',  '/srv/modoboa/env/lib/python2.7/lib-tk',  '/srv/modoboa/env/lib/python2.7/lib-old',  '/srv/modoboa/env/lib/python2.7/lib-dynload',  '/usr/lib/python2.7',  '/usr/lib/python2.7/plat-x86_64-linux-gnu',  '/usr/lib/python2.7/lib-tk',  '/srv/modoboa/env/local/lib/python2.7/site-packages',  '/srv/modoboa/env/lib/python2.7/site-packages']
Sat, 19 Jan 2019 10:53:48 +0100
```

**Comment by cremesk** [comment]
@tonioo you have an idea?

**Comment by cremesk** [comment]
Error reading /srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/../frontend/webpack-stats.json

No '../frontend/..' Directory on my disk. But i can find this file in:
/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/static/modoboa_radicale/webpack-stats.json

**Comment by tonioo** [comment]
@cremesk Yes, that's a know issue... What you can do is replace this line https://github.com/modoboa/modoboa-radicale/blob/master/modoboa_radicale/settings.py#L24 by the path you mentioned and it should work. The settings.py file you be located here: /srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/settings.py

**Comment by cremesk** [comment]
@tonioo Thanks! With Line 24:
`"STATS_FILE": "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/static/modoboa_radicale",`

works the Calendar-Site without Errors. However, it is still not possible to create a calendar.

`-rw-r--r-- 1 modoboa modoboa 2.6K Jan 16 18:57 /srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/static/modoboa_radicale/webpack-stats.json`

**Comment by tonioo** [comment]
@cremesk Ok, so now you should be able to see more information using a javascript debugger. (Look at the network tab and check server's response).

**Comment by cremesk** [comment]
Error in Debugger Console:
```
Source map error: request failed with status 404
Resource URL: https://mail.secnux.net/sitestatic/bootstrap/dist/css/bootstrap.min.css
Source Map URL: bootstrap.min.css.map
```

Request-URL: https://mail.secnux.net/api/v1/user-calendars/
Methode: POST
Internal Server Error

Request headers:
```
Host: mail.secnux.net
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0
Accept: application/json, text/plain, */*
Accept-Language: de,en-US;q=0.7,en;q=0.3
Accept-Encoding: gzip, deflate, br
Referer: https://mail.secnux.net/calendars/
X-CSRFTOKEN: uSgPUDi5ZCGlq9At0f1oJERNr57LH1NDv01UztNHqH673GGF0jUHt3iCePq2E6xS
Content-Type: application/json;charset=utf-8
X-Requested-With: XMLHttpRequest
Content-Length: 14
DNT: 1
Connection: keep-alive
Cookie: csrftoken=uSgPUDi5ZCGlq9At0f1oJERNr57LH1NDv01UztNHqH673GGF0jUHt3iCePq2E6xS; sessionid=uq8dbe0z4llp8d2bovy5accr0bnx9y6m
```

Response headers:
```
HTTP/1.1 500 Internal Server Error
Server: nginx/1.10.3
Date: Mon, 25 Feb 2019 10:56:53 GMT
Content-Type: text/plain
Content-Length: 20780
Connection: keep-alive
Vary: Accept-Language, Cookie
X-Frame-Options: SAMEORIGIN
Content-Language: en
```

**Comment by tonioo** [comment]
Don't you see anything in the response body tab?

**Comment by cremesk** [comment]
```

Traceback:  

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/utils/decorators.py" in inner
  185.                     return func(*args, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/utils/decorators.py" in inner
  185.                     return func(*args, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/django/views/decorators/csrf.py" in wrapped_view
  58.         return view_func(*args, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/viewsets.py" in view
  90.             return self.dispatch(request, *args, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  489.             response = self.handle_exception(exc)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/views.py" in handle_exception
  449.             self.raise_uncaught_exception(exc)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  486.             response = handler(request, *args, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/mixins.py" in create
  21.         self.perform_create(serializer)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/mixins.py" in perform_create
  26.         serializer.save()

File "/srv/modoboa/env/local/lib/python2.7/site-packages/rest_framework/serializers.py" in save
  214.             self.instance = self.create(validated_data)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/serializers.py" in create
  43.         self.create_remote_calendar(calendar)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/serializers.py" in create_remote_calendar
  21.         backend.create_calendar(calendar.encoded_url)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/backends/caldav_.py" in create_calendar
  73.         self.client.mkcalendar(url)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/caldav/davclient.py" in mkcalendar
  201.         return self.request(url, "MKCALENDAR", body)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/caldav/davclient.py" in request
  247.                              auth=auth, verify=self.ssl_verify_cert)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/requests/api.py" in request
  60.         return session.request(method=method, url=url, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/requests/sessions.py" in request
  533.         resp = self.send(prep, **send_kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/requests/sessions.py" in send
  646.         r = adapter.send(request, **kwargs)

File "/srv/modoboa/env/local/lib/python2.7/site-packages/requests/adapters.py" in send
  514.                 raise SSLError(e, request=request)

Exception Type: SSLError at /api/v1/user-calendars/
Exception Value: HTTPSConnectionPool(host='mail.secnux.net', port=443): Max retries exceeded with url: /radicale/sven.kinne%40secnux.net/dev (Caused by SSLError(SSLError(1, u'[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:661)'),))
Request information:
USER: sven.kinne@secnux.net
```

**Comment by tonioo** [comment]
@cremesk Can you try to update the pyOpenSSL and certifi packages:
```shell
> sudo -u modoboa
> source env/bin/activate
> pip install -U pyOpenSSL certifi
> exit
> sudo systemctl reload uwsgi
```

**Comment by cremesk** [comment]
same error after do this :/

**Comment by cremesk** [comment]
**Summary:**
Calendar settings.py need a modification on line 24:
```
From:
"STATS_FILE": CALENDAR_STATS_FILES.get("dev" if DEBUG else "prod"),
To:
"STATS_FILE": "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_radicale/static/modoboa_radicale/webpack-stats.json",
```

Same for Contacts:
```
"STATS_FILE": CONTACTS_STATS_FILES.get("dev" if DEBUG else "prod"),
"STATS_FILE": "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa_contacts/static/modoboa_contacts/webpack-stats.json",
```

SSL Problem:
> 
> So, I've found a quick fix but I think you have a SSL configuration issue somewhere.
> I've modified the /etc/hosts file by adding an entry for mail.secnux.net (so it is resolved as 127.0.0.1).
> Modoboa uses this value to connect to Radicale and send a calendar creation request but by default, it is resolved as 82.216.111.26. So, it goes out, then in, to send the creation request but it fails during SSL handshake step for an unknown reason...
